### DISCUSSION TITLE
colorizeModelLine does not work if you try to use immediately after creating the model

### DISCUSSION THREAD

**Discussion by davicatunda** [microsoft/monaco-editor#3649] [Q&A]
The code bellow using `monaco.editor.createModel` and  `monaco.editor.colorizeModelLine` results into a non-colorized line, but the second time we run (on this case with a setTimeout) it works correctly.

**monaco-editor version:** 0.28
**Browser:** Chrome
**OS:** Windows
**Playground code that reproduces the issue:**
On https://microsoft.github.io/monaco-editor/playground.html try
```js
const code = `/* Some example CSS */`
const model = monaco.editor.createModel(code, 'css');
const element = document.getElementById('root');
element.innerHTML = monaco.editor.colorizeModelLine(model, 1);
window.setTimeout(() => {
    element.innerHTML = monaco.editor.colorizeModelLine(model, 1);
}, 1000)
```

- the same happens if I tokenize right away instead `monaco.editor.tokenize(code, 'css')[0][0]);`.
- the same happens if we create an editor with `monaco.editor.create`

it appears that we need to listen to some event to guarantee that the editor/model is ready, is that the case?

**Comment by alexdima** [comment]
Yes, the only ones that work really well are `colorizeElement` or `colorize` as they return a promise when the language has finished loading. The `colorizeModelLine` one will colorize the model's line using the current state of things (i.e. as if the file would be displayed in an editor).
